### PR TITLE
fix(Realtime Composer): existing selection and insert issues

### PR DESCRIPTION
**Pull request by MartinSchoeler**
## Proposed changes (including videos or screenshots)

Fixes the following issues:
- Wrong insert position when adding emojis from the picker or pasting text
- When text was selected and a markdown tag applied (ctrl + b) if the text was at the end of the phrase, the composer would add a line break and duplicate the last character at the end of the selection
- When applying markdown tag, some selected newlines were removed
- When inserting emojis or mentions into an empty composer, nothing would happen
- Inconsistent caret and insert behavior after applying a markdown tag

Also removed unnecessary code with adjustments to the old gazzodown implementation

## Issue(s)

[RTC-12](https://rocketchat.atlassian.net/browse/RTC-12)
## Further comments

<a href="https://cubic.dev/pr/RocketChat/Rocket.Chat/pull/41417?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

[RTC-12]: https://rocketchat.atlassian.net/browse/RTC-12?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved caret/selection accuracy and round-tripping in the rich-text composer across formatting, inline elements, and multiline/newline edge cases.
  * Fixed text replace/insert behavior to maintain correct caret placement even when browser `execCommand` results in a no-op, using reliable fallbacks.
  * Enhanced wrapping to preserve blank lines and trailing newlines at boundary/end-line scenarios.
  * Restored the cursor/selection when refocusing the composer, and always sends the raw composer text.
  * Corrected final rendered output after link restoration, including placeholder line breaks.
* **Tests**
  * Expanded Jest coverage for caret round-tripping, selection mapping, and wrapping/editing edge cases.

### CODE REVIEW

File: apps/meteor/app/ui-message/client/messageBox/renderComposerMarkup.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	const input = document.createElement('div');
+	input.innerHTML = renderComposerMarkup(parse(text, {}), {});
+	document.body.appendChild(input);
+	return input;
+};
+
+const flatLength = (input: HTMLDivElement): number => {
+	setSelectionRange(input, Number.MAX_SAFE_INTEGER, Number.MAX_SAFE_INTEGER);
+	return getSelectionRange(input).selectionStart;
+};
```
[the remark this answers is not in the document]
Comment (MartinSchoeler) [reply apps/meteor/app/ui-message/client/messageBox/renderComposerMarkup.spec.tsx]: Explain that vague "An offset-map bug" please

**Review by cardoso — APPROVED** [review]
